src/routing/bootstrap.rs

```diff
@@ -137,11 +137,15 @@ impl<'a> State<'a> {
         mut bootstrap_addrs: Vec<SocketAddr>,
         relocate_details: Option<&SignedRelocateDetails>,
     ) -> Result<(Prefix, bls::PublicKey, BTreeMap<XorName, SocketAddr>)> {
+        // Avoid sending more than one request to the same peer.
+        let mut used_addrs = HashSet::new();
+
         loop {
+            used_addrs.extend(bootstrap_addrs.iter().copied());
             self.send_get_section_request(mem::take(&mut bootstrap_addrs), relocate_details)
                 .await?;
 
-            let (response, sender) = self.receive_get_section_response().await?;
+            let (response, sender) = self.receive_get_section_response(relocate_details).await?;
```

Comment:
just to confirm. so this is to resolve a bug previous that:
keeps contact old section to bootstrap even got relocated to new section.
right?

Reply:
The bug was that as a relocated node, we were checking the response against our current name, but we should have been checking it against the destination name instead. So we were incorrectly rejecting valid responses.